lib.ml

```diff
@@ -97,47 +103,70 @@ let string_to_xml : Value.t -> Value.t = function
 
 (* The following functions expect 1 argument. Assert false otherwise. *)
 let char_test_op fn pure =
-  (`PFun (fun args ->
+  (`PFun (fun args _ ->
       match args with
         | [c] -> (`Bool (fn (unbox_char c)))
         | _ -> assert false),
    datatype "(Char) ~> Bool",
    pure)
 
 let char_conversion fn pure =
-  (`PFun (fun args ->
+  (`PFun (fun args _ ->
       match args with
         | [c] -> (box_char (fn (unbox_char c)))
         | _ -> assert false),
    datatype "(Char) -> Char",
    pure)
 
 let float_fn fn pure =
-  (`PFun (fun args ->
+  (`PFun (fun args _ ->
       match args with
         | [c] -> (box_float (fn (unbox_float c)))
         | _ -> assert false),
    datatype "(Float) -> Float",
   pure)
 
 let p1 fn =
-  `PFun (fun args ->
+  `PFun (fun args _ ->
```

Comment:
You could define p1D etc. first and then define p1, p2, p3 as special cases e.g. p1 fn = p1d (fun _ x -> fn)